wordpress-mobile/WordPress-iOS
# iPad: Notifications splitview

This PR introduces a split view to Notifications, following on from @kurzee's readability work. Hopefully everything should work as expected. The left pane always shows the notifications list, and the right pane shows the detail for the selected notification.

Things of note:

* I opted to disable full screen for reader posts in Notifications. Firstly because it just felt weird to me to use. I primarily find notifications to be something you skim through, rather than spend time in reading posts. Secondly, because the navigation is somewhat different to Reader and we'd need to do some design work to figure out the best way to integrate it: in Reader, you're always pushing a fullscreen view onto the detail stack, whereas in notifications the fullscreen view would be set as the root of the detail stack. We could always revisit this in the future.
* If the user has no notifications at all (you can test this with a brand new account), we're showing a full screen "No notifications" screen, much like you get if you have no blogs at all.

To test:

* Ensure you test on different devices, in various multitasking states. 
* Ensure that if you move from the split view being collapsed to expanded, that a notification gets selected.
* Ensure that new accounts both a) show a correct 'no notifications' screen, and b) transition correctly to show notifications once one is received.
* Check state restoration – both the selected segment index and the selected notification should be restored. Also check what happens if a new notification arrives whilst the app isn't running.
* Check that the navigation buttons in the reader detail screens work correctly, and that if you transition back to an expanded split view that the correct notification is highlighted.
* Ensure that full screen doesn't activate at any point!

Needs review: @kurzee
